lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -4179,8 +4179,15 @@ static LogicalResult getOutputTypeAndPoolingParameters(
   outputTy = getOutputTypeForNonAdaptivePoolingOp<AtenOpT, tosaOp>(
       inputTy, kernelSizeInts, strideInts, paddingInts, dilationArray,
       ceilMode);
-  padArr[1] = padArr[1] + paddingInts[0];
-  padArr[3] = padArr[3] + paddingInts[1];
+  RankedTensorType outputTensorTy = outputTy.cast<RankedTensorType>();
+  auto inH = inputTy.getShape()[2];
+  auto inW = inputTy.getShape()[3];
+  auto outH = outputTensorTy.getShape()[1];
+  auto outW = outputTensorTy.getShape()[2];
+  padArr[1] = (outH - 1) * strideInts[0] - inH - padArr[0] + kernelSizeInts[0];
+  padArr[1] = (padArr[1] < padArr[0]) ? padArr[0] : padArr[1];
```

Comment:
no, the code is not for swapping. It's to compute the round up integer number since torch has the same padding value for left and right, top and bottom.